app/services/terms_of_use/acceptor.rb

```diff
@@ -38,10 +42,16 @@ def terms_of_use_agreement
       @terms_of_use_agreement ||= user_account.terms_of_use_agreements.new(agreement_version: version)
     end
 
-    def update_sign_up_service
+    def asynchronous_update
+      terms_of_use_agreement.accepted!
       SignUpServiceUpdaterJob.perform_async(attr_package_key)
     end
 
+    def synchronous_update
+      SignUpServiceUpdaterJob.new.perform(attr_package_key)
```

Comment:
I guess we will have to create the ToU agreement first. Inside the job it expects the ToU to have been accepted before accepting with MAP SuS

Reply:
Cause when I test this I get: `app/sidekiq/terms_of_use/sign_up_service_updater_job.rb:44:in `terms_of_use_agreement'
NoMethodError: undefined method `accepted?' for nil:NilClass`

Reply:
oh yeahhh. womp womp

Reply:
I changed it to this:
```ruby
def synchronous_update
  terms_of_use_agreement.accepted!
  SignUpServiceUpdaterJob.new.perform(attr_package_key)
rescue
  terms_of_use_agreement.destroy
  raise
end
```
good with that?